es-ds-components/components/es-video.vue

```diff
@@ -0,0 +1,73 @@
+<script setup lang="ts">
+import { useElementSize } from '@vueuse/core';
+
+interface IProps {
+    altText: string;
+    coverImageUrl: string;
+    embedUrl: string;
+}
+const props = defineProps<IProps>();
+const showVideo = ref(false);
+function getVideoIdFromUrl(url: string) {
+    const embedPattern = /https:\/\/www\.youtube\.com\/embed\/(?<videoId>[A-Za-z0-9-]+)/;
+    const parsedUrl = embedPattern.exec(url);
+    if (parsedUrl) {
+        return parsedUrl?.groups?.videoId;
+    }
+    return '';
+}
+const videoId = getVideoIdFromUrl(props.embedUrl);
+const embedUrlWithParams = `https://www.youtube.com/embed/${videoId}?rel=0&autoplay=1&cc_load_policy=1&cc_lang_pref=en`;
```

Comment:
Because these two variables depend on props, I think we may need to make them [computed](https://vuejs.org/guide/essentials/computed.html) so they can be reactive and react if the prop values change.